lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -0,0 +1,641 @@
+        """Manager of OpenSearch backup relations."""
+        super().__init__(charm, S3_RELATION)
+        self.charm = charm
+        # s3 relation handles the config options for s3 backups
+        self.s3_client = S3Requirer(self.charm, S3_RELATION)
+        self.framework.observe(self.charm.on[S3_RELATION].relation_departed, self.on_s3_depart)
+        self.framework.observe(self.s3_client.on.credentials_changed, self.on_s3_change)
+        self.framework.observe(self.charm.on.create_backup_action, self._on_create_backup_action)
+        self.framework.observe(self.charm.on.list_backups_action, self._on_list_backups_action)
+
+    @property
+    def _plugin(self) -> OpenSearchPlugin:
+        return self.charm.plugin_manager.get_plugin(OpenSearchBackupPlugin)
+
+    def _on_list_backups_action(self, event: ActionEvent) -> None:
+        """Returns the list of available backups to the user."""
+        try:
+            self.apply_post_restart_if_needed()
+        except OpenSearchPluginRelationClusterNotReadyError:
+            event.fail(
+                "Failed: cluster must be active before applying the remaining backup config"
+            )
+            return
+
+        if not self._check_action_can_run(event):
+            return
+        backups = {}
+        try:
+            backups = self._list_backups()
+        except OpenSearchBackupNeworkError as e:
+            event.fail(f"Failed: network error {e}")
+            return
+        except OpenSearchListBackupError():
+            event.fail("Failed: listing backups error, please check logs")
+            return
+        event.set_results({"snapshots": (json.dumps(backups)).replace("_", "-")})
+
+    def _on_create_backup_action(self, event: ActionEvent) -> None:
+        """Creates a backup from the current cluster."""
+        try:
+            self.apply_post_restart_if_needed()
+        except OpenSearchPluginRelationClusterNotReadyError:
+            event.fail(
+                "Failed: cluster must be active before applying the remaining backup config"
+            )
+            return
+
+        if not self.charm.unit.is_leader():
+            event.fail("Failed: the action can be run only on leader unit")
+            return
+
+        if not self._check_action_can_run(event):
+            return
+
+        new_backup_id = None
+        try:
+            # Check if any backup is not running already, or RetryError happens
+            self.wait_backup_success()
+            # gets the latest value
+            # the very firs time this action is executed, _list_backups returns empty
+            # hence, the "or [0]"
+            new_backup_id = int(max(self._list_backups().keys() or [0])) + 1
+            output = self.get_service_status(
+                self._request(
+                    "PUT",
+                    f"_snapshot/{OPENSEARCH_REPOSITORY_NAME}/{new_backup_id}?wait_for_completion=true",
+                    payload={
```

Comment:
So, I am not sure if we want to take a snapshot of indices such as `.opensearch-security`, which contains sensitive information for that specific cluster. Otherwise, yes we can use the defaults.